src/kafka-connect/view/view.controller.js

```diff
@@ -86,74 +71,27 @@ angularAPP.controller('ConnectorDetailCtrl', function ($rootScope, $scope, $rout
   };
 
   $scope.toggleEditor = function () {
-    $scope.connectorConfigurationEditable = !$scope.connectorConfigurationEditable;
-    if (!$scope.connectorConfigurationEditable) {
-        $scope.aceBackgroundColor = "rgba(0, 128, 0, 0.04)";
-        $scope.showSyntaxValidation = true;
+    $scope.isEditing = !$scope.isEditing;
+
+    if ($scope.isEditing) {
+        $scope.model = angular.copy($scope.connectorDetails.config); // clone dirty model
     } else {
-        $scope.aceBackgroundColor = "#fff";
-        $scope.validConfig = "";
-        $scope.showSyntaxValidation = false;
-        invalidateEditorMarkers();
         $scope.errorConfigs = null;
     }
   };
 
   $scope.cancelEditor = function() {
-    $scope.connectorDetails.connectorDetailsInString = $scope.connectorDetailsInStringBefore;
+    $scope.model = $scope.connectorDetails.config; // replace dirty model with pristine
+    $scope.form.$setPristine();
     $scope.toggleEditor();
   }
 
-  $scope.aceLoaded = function (_editor) {
-    console.log("Ace loaded");
-    $scope.editor = _editor;
-    $scope.editor.$blockScrolling = Infinity;
-    $scope.acePropertyFileSession = _editor.getSession();
-    var lines = $scope.connectorDetails.connectorDetailsInString.split("\n").length;
-    _editor.setOptions({
-      minLines: lines,
-      maxLines: lines,
-      highlightActiveLine: false
-    });
-  };
-
-  $scope.aceChanged = function (_editor) {
-    $scope.editor = _editor;
-    $scope.editor.$blockScrolling = Infinity;
-    var aceContent = $scope.acePropertyFileSession.getDocument().getValue();
-    $scope.connectorDetails.connectorDetailsInString = aceContent;
-  };
-
-  var  markerRange;
-  $scope.aceValidate = function (_editor) {
-      var Range = ace.require('ace/range').Range;
-      invalidateEditorMarkers();
-      var errorLines = calculateErrorLines();
-      $scope.errorIds = [];
-      angular.forEach(errorLines, function (errorline) {
-        markerRange = new Range(errorline, 1, errorline, 2);
-        markerRange.id =  $scope.acePropertyFileSession.addMarker( markerRange, "myMarker", "fullLine", "ace_error");
-        $scope.errorIds.push(markerRange.id );
-      });
-   };
```

Comment:
This Ace editor validation functionality did not really seem to be used at the moment, so I removed it for simplicity. JSON syntax is still [being validating](https://github.com/Landoop/kafka-connect-ui/pull/61/files#diff-7e6c69b7e88c23526427ea8f94a32228R48) in `configurationEditorJson`.

I like this idea though: Adding Ace editor error markers based on custom validation messages. This would be a neat enhancement to add with some validation refactoring work.